src/machine/machine_mimxrt1062_i2c.go

```diff
@@ -0,0 +1,646 @@
+		if result := i2c.controllerReceive(r); resultSuccess != result {
+			return errI2CReadTimeout
+		}
+	}
+
+	// generate stop condition on bus
+	if result := i2c.stop(); resultSuccess != result {
+		return errI2CSignalStopTimeout
+	}
+
+	return nil
+}
+
+// WriteRegister transmits first the register and then the data to the
+// peripheral device.
+//
+// Many I2C-compatible devices are organized in terms of registers. This method
+// is a shortcut to easily write to such registers. Also, it only works for
+// devices with 7-bit addresses, which is the vast majority.
+func (i2c I2C) WriteRegister(address uint8, register uint8, data []byte) error {
+	option := transferOption{
+		flags:          transferDefault,  // transfer options bit mask (0 = normal transfer)
+		peripheral:     uint16(address),  // 7-bit peripheral address
+		direction:      directionWrite,   // directionRead or directionWrite
+		subaddress:     uint16(register), // peripheral sub-address (transferred MSB first)
+		subaddressSize: 1,                // byte length of sub-address (maximum = 4 bytes)
+	}
+	if result := i2c.controllerTransferPoll(option, data); resultSuccess != result {
+		return errI2CWriteTimeout
+	}
+	return nil
+}
+
+// ReadRegister transmits the register, restarts the connection as a read
+// operation, and reads the response.
+//
+// Many I2C-compatible devices are organized in terms of registers. This method
+// is a shortcut to easily read such registers. Also, it only works for devices
+// with 7-bit addresses, which is the vast majority.
+func (i2c I2C) ReadRegister(address uint8, register uint8, data []byte) error {
+	option := transferOption{
+		flags:          transferDefault,  // transfer options bit mask (0 = normal transfer)
+		peripheral:     uint16(address),  // 7-bit peripheral address
+		direction:      directionRead,    // directionRead or directionWrite
+		subaddress:     uint16(register), // peripheral sub-address (transferred MSB first)
+		subaddressSize: 1,                // byte length of sub-address (maximum = 4 bytes)
+	}
+	if result := i2c.controllerTransferPoll(option, data); resultSuccess != result {
+		return errI2CWriteTimeout
+	}
+	return nil
+}
+
+func (i2c *I2C) reset(freq uint32) {
+	// disable interface
+	i2c.Bus.MCR.ClearBits(nxp.LPI2C_MCR_MEN)
+
+	// software reset all interface registers
+	i2c.Bus.MCR.Set(nxp.LPI2C_MCR_RST)
+
+	// RST remains set until manually cleared!
+	i2c.Bus.MCR.ClearBits(nxp.LPI2C_MCR_RST)
+
+	// disable host request
+	i2c.Bus.MCFGR0.Set(0)
+
+	// enable ACK, use I2C 2-pin open drain mode
+	i2c.Bus.MCFGR1.Set(0)
+
+	// set FIFO watermarks (RX=1, TX=1)
+	mfcr := (uint32(0x1) << nxp.LPI2C_MFCR_RXWATER_Pos) & nxp.LPI2C_MFCR_RXWATER_Msk
+	mfcr |= (uint32(0x1) << nxp.LPI2C_MFCR_TXWATER_Pos) & nxp.LPI2C_MFCR_TXWATER_Msk
+	i2c.Bus.MFCR.Set(mfcr)
+
+	// configure clock using receiver frequency
+	i2c.setFrequency(freq)
+
+	// clear reset, and enable the interface
+	i2c.Bus.MCR.Set(nxp.LPI2C_MCR_MEN)
```

Comment:
When I tried it in drivers/examples/ssd1306/i2c_128x64, the transmission immediately after initialization disappeared.
Perhaps a wait loop like the following is needed.
Timeout may also be needed.

```suggestion
	i2c.Bus.MCR.Set(nxp.LPI2C_MCR_MEN)

	// wait for the I2C bus to idle
	for {
		if i2c.Bus.MSR.Get()&nxp.LPI2C_MSR_BBF == 0 {
			break
		}
	}
```

Reply:
OK, looks great to me. Reworded slightly:

```go
for i2c.Bus.MSR.Get()&nxp.LPI2C_MSR_BBF != 0 {
}
```